firebase/rules/tests/invocations.spec.ts

```diff
@@ -0,0 +1,136 @@
+import 'jest';
+import * as matchers from '../matchers';
+import * as targaryen from 'targaryen';
+
+import { SYSTEM_USER, Invocation, InvocationType } from '@machinelabs/models';
+
+const anonymousUser = targaryen.util.users.anonymous;
+const rules = require('../database.rules.json');
+
+const dummyInvocationLab = {
+  id: '1',
+  directory: 'dummy/directory'
+};
+
+const dummyInvocationExecution = {
+  execution_id: '1'
+};
+
+const dummyInvocation: Invocation = {
+  id: '1',
+  timestamp: Date.now(),
+  user_id: null,
+  type: Date.now() % 2 ? InvocationType.StartExecution : InvocationType.StopExecution,
+  data: Date.now() % 2 ? dummyInvocationLab : dummyInvocationExecution
+};
+
+describe('/invocations', () => {
+  let currentUser;
+  let systemUser;
+
+  let dummyInvocationCurrentUser: Invocation;
+  let dummyInvocationSystemUser: Invocation;
+
+  beforeEach(() => {
+    jest.addMatchers(matchers);
+
+    targaryen.util.setFirebaseData({
+      users: {
+        '1': {
+          common: {
+            id: '1',
+            displayName: 'Foo Bar',
+            email: 'foo.bar@example.com',
+            bio: '',
+            isAnonymous: false,
+            photoUrl: 'avatars.com/1'
+          }
+        },
+        '2': {
+          common: {
+            id: '2',
+            displayName: 'Fizz Buzz',
+            email: 'fizz.buzz@example.com',
+            bio: '',
+            isAnonymous: false,
+            photoUrl: 'avatars.com/2'
+          }
+        }
+      },
```

Comment:
We can remove these two user objects here because they aren't needed to test the rules for invocations.